app/model/view/GroupViewFactory.php

```diff
@@ -164,8 +175,12 @@ public function getGroup(Group $group): array {
       "parentGroupId" => $group->getParentGroup() ? $group->getParentGroup()->getId() : null,
       "parentGroupsIds" => $group->getParentGroupsIds(),
       "childGroups" => [
-        "all" => $group->getChildGroupsIds(),
-        "public" => $group->getPublicChildGroupsIds()
+        "all" => $childGroups->map(function (Group $group) { return $group->getId(); })->getValues(),
+        "public" => $publicChildGroups->map(function (Group $group) { return $group->getId(); })->getValues()
+      ],
+      "archivedChildGroups" => [
```

Comment:
Similarly to the user entity situation, I believe that this is unnecessary and it will bite us in a few years. At present, we have 60 groups and the ReCodEx is not fully used right now. In the future, we can have easily 150 new groups each semester. That means after few years, we could easily get up to 1000 records being sent over. Archived data should be requested explicitly.